eng/pipelines/templates/steps/analyze.yml

```diff
@@ -0,0 +1,22 @@
+parameters:
+  - name: RustToolchain
+    type: string
+    default: stable
+
+steps:
+  - template: /eng/pipelines/templates/steps/use-rust.yml@self
+    parameters:
+      Toolchain: ${{ parameters.RustToolchain }}
+
+  - pwsh: |
+      . ./eng/common/scripts/Helpers/CommandInvocation-Helpers.ps1
+
+      $env:RUSTDOCFLAGS = "-D warnings"
+      $env:RUSTFLAGS = "-Dwarnings"
+
+      Invoke-LoggedCommand "cargo +$(RustToolchain) check -p azure_core --no-default-features"
+      Invoke-LoggedCommand "cargo +$(RustToolchain) install cargo-readme"
+      Invoke-LoggedCommand "cargo +$(RustToolchain) fmt --all -- --check"
+      Invoke-LoggedCommand "cargo +$(RustToolchain) clippy --all"
+      Invoke-LoggedCommand "cargo +$(RustToolchain) doc --all --no-deps"
```

Comment:
You should also run `eng/scripts/verify-dependencies.rs` (source-as-script). You'd either need to run it as any command in bash, or pass the shebang command at the top of the file in pwsh.

Reply:
added

Reply:
Added to test-packages.ps1 as it needs to be run in each affected package